pkg/reconciler/contour/resources/httpproxy.go

```diff
@@ -300,8 +300,16 @@ func MakeHTTPProxies(ctx context.Context, ing *v1alpha1.Ingress, serviceToProtoc
 				}
 
 				hostProxy.Name = kmeta.ChildName(ing.Name+"-"+class+"-", host)
-				hostProxy.Spec.VirtualHost = &v1.VirtualHost{
-					Fqdn: host,
+
+				if cfg.Contour.CORSPolicy != nil {
+					hostProxy.Spec.VirtualHost = &v1.VirtualHost{
+						Fqdn:       host,
+						CORSPolicy: cfg.Contour.CORSPolicy,
+					}
+				} else {
+					hostProxy.Spec.VirtualHost = &v1.VirtualHost{
+						Fqdn: host,
+					}
 				}
```

Comment:
Don't need such a complex if statement

```suggestion
				hostProxy.Spec.VirtualHost = &v1.VirtualHost{
					Fqdn:       host,
				}

				if cfg.Contour.CORSPolicy != nil {
					hostProxy.Spec.VirtualHost.CORSPolicy = cfg.Contour.CORSPolicy
				}
```